### PR TITLE
fix: static rule 53 and lookup rule 58

### PR DESCRIPTION
**Pull request by andrewmorris43**
## Description
- Fixes rule 53.CurrentPostingAndPrimaryCareProvider always failing
- Modifies rule 58.CurrentPosting to allow null currentPosting

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs [hunk 1]
```diff
@@ -83,6 +83,7 @@ public bool ValidateOutcode(string postcode)
     /// <returns>bool, whether or not the current posting is valid.<returns>
     public bool ValidateCurrentPosting(string currentPosting)
     {
+        if (currentPosting == null) return true;
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs:86]: if the rule is supposed to pass if the current posing is null, change this in the rules to do string.IsNullOrEmty(currentposting) OR validateCurrentPosting.

I think it is better to have most of the logic in the rules

Comment (andrewmorris43) [reply application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs]: Yeah agree with this. I've moved this to the rule and added some unit tests too.

**Review by Warren-Pitterson — APPROVED** [review]

File: application/CohortManager/src/Functions/Shared/Common/ValidationHelper.cs [hunk 2]
```diff
@@ -36,41 +36,44 @@ public static bool ValidatePastDate(string dateString)
 
     public static bool ValidateNHSNumber(string nhsNumber)
     {
-            // Check the NHS number is a number
```
Comment (SamRobinson75684) [inline application/CohortManager/src/Functions/Shared/Common/ValidationHelper.cs:39]: can you make this method a little more concise? it seems to be doing more than one check in the same method. Could split out this method so each of the methods it uses has a single responsibility

Comment (andrewmorris43) [reply application/CohortManager/src/Functions/Shared/Common/ValidationHelper.cs:39]: This is just some file formatting changes in here, I haven't actually made any code changes in this file. I think your suggestion would be better made on a separate PR.

**Review by SamRobinson75684 — APPROVED** [review]

**Pull request merged**